Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificatesExtensions.kt

```diff
@@ -85,20 +85,34 @@ private fun Collection<CwaCovidCertificate>.rule2FindRecentRaCertificate(
  */
 private fun Collection<CwaCovidCertificate>.rule3FindRecentLastShot(
     nowUtc: Instant
-): CwaCovidCertificate? = this
-    .filterIsInstance<VaccinationCertificate>()
-    .filter {
-        with(it.rawCertificate.vaccination) { doseNumber == totalSeriesOfDoses }
+): CwaCovidCertificate? {
+    val twoDoseVaccines = listOf("EU/1/20/1528", "EU/1/21/1529", "EU/1/20/1507")
+    val isOlderThanTwoWeeks = { certificate: VaccinationCertificate ->
+        Days.daysBetween(
+            certificate.rawCertificate.vaccination.vaccinatedOn,
+            nowUtc.toLocalDateUtc()
+        ).days > 14
     }
-    .filter {
-        Days.daysBetween(it.rawCertificate.vaccination.vaccinatedOn, nowUtc.toLocalDateUtc()).days > 14
-    }
-    .maxWithOrNull(
-        compareBy(
-            { it.rawCertificate.vaccination.vaccinatedOn },
-            { it.headerIssuedAt }
+    return this
+        .filterIsInstance<VaccinationCertificate>()
+        .filter { with(it.rawCertificate.vaccination) { doseNumber == totalSeriesOfDoses } }
+        .filter {
+            with(it.rawCertificate.vaccination) {
+                when {
+                    totalSeriesOfDoses > 2 && twoDoseVaccines.contains(medicalProductId) -> true
+                    totalSeriesOfDoses == 2 && medicalProductId !in twoDoseVaccines -> true
+                    totalSeriesOfDoses == 1 && twoDoseVaccines.contains(medicalProductId) -> true
```

Comment:
can you check if this case is also covered https://github.com/corona-warn-app/cwa-app-android/issues/4334?

Reply:
Alright

Reply:
I did not mean the booster rule warning. is not the user has booster vaccination 2/2 and he had already 1/1. I expect that the booster 2/2 should have priority, is not that should be covered by this ticket point one in the AC

Reply:
Only if the 2/2 vaccine is a J&J. Otherwise, from the acceptance criteria it looks like it will take priority after 14 days. Tech spec mentions only J&J too https://github.com/corona-warn-app/cwa-app-tech-spec/pull/102/files#diff-25c8b7188514d7cdfb430f7f8f963fed63bca279fd74753bc6cefe29dd599b55

Reply:
I see

Reply:
@mtwalli #4334 is about a Booster Notification, not about priority.

Reply:
@mlenkeit  ,Yes, there are two things in the issue . The booster warning information and the priority 
> Validation check should detect, that this vaccination is the second in a heterologous vaccination scheme and therefore consider the certificate to be valid (help text implies that these vaccinations are valid immediatly)

 but you can see that the user has Two vaccinations 1:1 and  2:2. 2:2 is not taking priority. I expected that should be covered by the changes in here ,but @SamuraiKek  said it is not since the 2:2 is not J&J Vaccine ,and therefore it will have priority after 14 days despite that it is a booster vaccination

Reply:
> it is not since the 2:2 is not J&J Vaccine ,and therefore it will have priority after 14 days despite that it is a booster vaccination

@mtwalli correct. for the priority rules, we also only take a look at an individual DCC and not at a set of DCCs. So when inspecting the 2/2 DCC for itself, we cannot say whether it's a booster or not, so we'll assume it is not and so it's subject to the 14 day period.